### PR TITLE
Starting to do unit test

### PR DESCRIPTION
**Pull request by emmohac**
@anthopark don't merge. It's in progress.

### CODE REVIEW

**Review by emmohac — COMMENTED** [review]
I think we need to create src/helper/{course,plan,helpers}.js and put the handler in there. So the structure will be something like this:
```
// src/helper/course.js
module.export getCourseSearch = async(req, res) => {}
// src/routers/course.js
const getCourseSearch = require('../helper/course')
router.get('/api/course/search', logRequest, getCourseSearch)
```
This helps clean up the router and allow us to unit test the handler, in this case is getCourseSearch.

File: backend/src/routers/course.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ router.get('/api/course/search', logRequest, async (req, res) => {
                 dept, level, num: { $regex: num, $options: "i" }
             });
         }
-
+        // should send statuscode.OK here too
```
Comment (emmohac) [inline backend/src/routers/course.js:49]: We need to send a success code for the successful case.

Comment (emmohac) [reply backend/src/routers/course.js:49]: @anthopark what do you think?

Comment (anthopark) [reply backend/src/routers/course.js:49]: @emmohac Yes, there should be sending statuscode.OK. thanks!

**Comment by anthopark** [comment]
> I think we need to create src/helper/{course,plan,helpers}.js and put the handler in there. So the structure will be something like this:
> 
> ```
> // src/helper/course.js
> module.export getCourseSearch = async(req, res) => {}
> // src/routers/course.js
> const getCourseSearch = require('../helper/course')
> router.get('/api/course/search', logRequest, getCourseSearch)
> ```
> 
> This helps clean up the router and allow us to unit test the handler, in this case is getCourseSearch.

I wanted the new `helper.js` containing "fetchCompletePlan" to be specific to the plan routers. And the module can have multiple helper functions in it.
Instead of `src/helper/{course,plan,helpers}`, do you think I should make it `src/routers/helpers/{plan, course}`?

File: backend/src/routers/course.js (hunk 1 above)
Comment (emmohac) [reply backend/src/routers/course.js:49]: OK I can work on that

**Comment by emmohac** [comment]
@anthopark it depends on how you want to structure the codebase. There are some reference that you can use to help you decide: https://www.coreycleary.me/project-structure-for-an-express-rest-api-when-there-is-no-standard-way

**Pull request closed without merging**